File: CivitasJava/src/Civitas/SorpresaEspeculador.java
package Civitas;
import java.util.ArrayList;

public class SorpresaEspeculador extends Sorpresa {
    private int valor;
    
    SorpresaEspeculador(int valor, String texto){
        super(texto);
        this.valor=valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            Especulador espc = new Especulador(todos.get(actual),valor);
            todos.remove(actual);
            todos.add(espc);
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa: { \n Texto = " + texto + "\n Tipo = Especulador" + 
                "\n Valor = " + valor + "}";
    }
}

File: CivitasRuby/lib/sorpresa_especulador.rb
# encoding:utf-8
require_relative 'sorpresa.rb'
require_relative 'especulador'
module Civitas
  class SorpresaEspeculador < Sorpresa
    def initialize(texto,valor)
      super(texto)
      @valor = valor
    end
    
    def informe(actual, todos)
      Diario.instance.ocurre_evento("Aplicando sorpresa " + self.to_s + 
        " al jugador " + todos[actual].nombre + " de tipo Especulador")
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual,todos)
        espc = Especulador.nuevo_especulador(todos[actual],@valor)
        todos.delete_at(actual)
        todos << espc
      end
    end
    
    def to_s
      "Sorpresa: { \n Texto: #{@texto}  \n Tipo: Especulador   \n Valor: #{@valor} }"
    end
    
  end
end
